app/models.py

```diff
@@ -59,3 +61,44 @@ class Tasks(db.Model):
 
     def __repr__(self):
         return '<Task {}>'.format(self.task)
+
+
+# Flask-Security
+
+roles_users = db.Table('roles_users',
+        db.Column('user_id', db.Integer(), db.ForeignKey('user.id')),
+        db.Column('role_id', db.Integer(), db.ForeignKey('role.id')))
+
+
+class Role(db.Model, RoleMixin):
+    id = db.Column(db.Integer(), primary_key=True)
+    name = db.Column(db.String(80), unique=True)
+    description = db.Column(db.String(255))
+
+
+class User(db.Model, UserMixin):
+    id = db.Column(db.Integer, primary_key=True)
+    email = db.Column(db.String(255), unique=True)
+    password = db.Column(db.String(255))
+    active = db.Column(db.Boolean())
+    confirmed_at = db.Column(db.DateTime())
+    roles = db.relationship('Role', secondary=roles_users, backref=db.backref('users', lazy='dynamic'))
+
+    # create a mentee account with the same email as our flask-security user
+    @validates('email')
+    def update_mentee(self, key, value):
+        alread_present = Mentees.query.filter_by(mentee_email=value).all()
```

Comment:
is `.exists()` equivalent not available for SQLAlchemy? if not, use `count` instead of `all`?

Reply:
why not add an exception for integrity constraint?
https://stackoverflow.com/questions/12265451/ask-forgiveness-not-permission-explain

Reply:
I'll see. That's a little hack that I did to keep `User` and `Mentee` separate yet when a user is created, also create a mentee (1-1 between User and Mentee). Similar to how django-user and LG-user were linked in our LG backend.

This doesn't look very right. Let me know if you have any suggestions on this? I tried researching about `triggers` but that involved writing raw SQL trigger queries. 

`ask forgiveness not permission` <-- amazing! thanks

Reply:
let's discuss this offline. in the meanwhile, create a eer diagram for me?

Reply:
Yes

Reply:
stay away from database triggers

sql = triggers
https://softwareengineering.stackexchange.com/questions/123074/sql-triggers-and-when-or-when-not-to-use-them

Django = signals
https://lincolnloop.com/blog/django-anti-patterns-signals/

Rails = callbacks
https://medium.com/planet-arkency/the-biggest-rails-code-smell-you-should-avoid-to-keep-your-app-healthy-a61fd75ab2d3

Reply:
Thanks. Yeah, they are ugly to see in code as well. I'm not touching them with a stick.